### PR TITLE
constraint chooser UI design

### PR DESCRIPTION
**Pull request by odanivan**
I added the constraint chooser component and designed it in the base modal style. Therefore I also gave the base modal component an optional width property to make it more customizable as well as an optional scroll bar within the base modal. The constraint chooser component lets the user choose to limit the licence by specifying what category is of need and by choosing the operator that is applied onto the input value. 
Here, the user needs to input a number (when limiting the age for example) or it's necessary to make a selection from a list of options (like selecting the states of germany in which the licence is granted - geographic limitation to the licence).

The next step is to implement the backbone that saves the input after clicking the "Annehmen" button and to pipe it through to the JSON file generator.

### CODE REVIEW

File: src/components/ActionChooser.vue [hunk 1]
```diff
@@ -31,67 +39,65 @@ export default {
...
-  
+
   data: function() {
-      return {
-          currentSelected: null,
-          currentAction: "",
-          actions: {
-              1: {id: 1, label: "Aggregieren"},
-              2: {id: 2, label: "Annotieren"},
-              4: {id: 4, label: "Annonymisieren"},
```
Comment (hkohlsaat) [inline src/components/ActionChooser.vue:42]: Es wäre besser URIs als id zu benutzen. Z.B. "http://ordl.org/annotieren" oder so

File: src/components/BaseModal.vue [hunk 2]
```diff
@@ -2,12 +2,12 @@
   <transition name="modal">
     <div class="modal-mask">
       <div class="modal-wrapper">
-        <div class="modal-container">
+        <div v-bind:style="{ width: this.width + 'px' }" class="modal-container">
```
Comment (hkohlsaat) [inline src/components/BaseModal.vue:5]: Geht das nicht im style Abschnitt?

File: src/components/BaseModal.vue [hunk 3]
```diff
@@ -54,8 +65,11 @@ export default {
   vertical-align: middle;
 }
 
+.scrollable {
+  overflow-y: scroll !important;
```
Comment (hkohlsaat) [inline src/components/BaseModal.vue:69]: nie !important nutzen

File: src/components/ConstraintChooser.vue [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        3: { id: 3, name: "<" },
+        4: { id: 4, name: ">" }
+      },
+      states: {
+        1: { id: 1, name: "Baden-Württemberg" },
+        2: { id: 2, name: "Bayern" },
+        3: { id: 3, name: "Berlin" },
+        4: { id: 4, name: "Brandenburg" },
+        5: { id: 5, name: "Bremen" },
+        6: { id: 6, name: "Hamburg" },
```
Comment (hkohlsaat) [inline src/components/ConstraintChooser.vue:100]: gibt's dafür keine URIs?

File: src/components/RuleItem.vue [hunk 5]
```diff
@@ -63,9 +63,10 @@ export default {
   display: inline-block;
 }
 
-.rule-header-container {
+.rule-header-container {
```
Comment (hkohlsaat) [inline src/components/RuleItem.vue:66]: trailing white space :O

**Review by hkohlsaat — APPROVED** [review]
nagut

File: src/components/BaseModal.vue (hunk 2 above)
Comment (odanivan) [reply src/components/BaseModal.vue:5]: Nein

File: src/components/BaseModal.vue (hunk 3 above)
Comment (odanivan) [reply src/components/BaseModal.vue:69]: außer in Ausnahmefällen

File: src/components/RuleItem.vue (hunk 5 above)
Comment (odanivan) [reply src/components/RuleItem.vue:66]: haha

**Pull request merged**